src/Components/Overlay/index.jsx

```diff
@@ -2,58 +2,169 @@ import React, { useEffect, useState } from "react";
       const url = DICTIONARY_URL + word;
-      let arrayOfMeanings = [];
       let { succeeded, response } = await fetchData(url);
       if (succeeded) {
-        setError({ ...error, isSuccess: true });
-        response.map((data) => {
-          return data.meanings.map((each) =>
-            each.definitions.map((desc) =>
-              arrayOfMeanings.push(desc.definition)
-            )
+        let simplifiedResponse = serializeWordDefinition(response);
+        let partOfSpeechArrayTemp = [];
+        simplifiedResponse?.sections?.forEach((data) => {
+          let partOfSpeechArrayEachSection = [];
+          data?.definitions.forEach((each) =>
+            partOfSpeechArrayEachSection.push(each.partOfSpeech)
           );
+          partOfSpeechArrayTemp.push(partOfSpeechArrayEachSection);
         });
-        setDefinition(arrayOfMeanings);
+        setPartOfSpeechArray(partOfSpeechArrayTemp);
+        setWordDetails(simplifiedResponse);
       } else {
-        setError({
-          isSuccess: false,
-          errorMessage: response.response.data.title,
+        setIsModalVisible(false);
+        notification.error({
+          message: "Error",
+          description: "Word not found!",
         });
       }
     };
     getWordDefinition();
   }, []);
 
+  const closeModal = () => {
+    setIsModalVisible(false);
+    document.querySelector(".ant-dropdown-menu").classList.remove("hidden");
+  };
+
   return (
     <>
-      <div
-        className="overlay-backdrop"
-        onClick={() => setIsOverlayVisible(false)}
+      <Modal
+        title={word}
+        visible={isModalVisible}
+        onOk={closeModal}
+        onCancel={closeModal}
+        okText="Close"
+        cancelButtonProps={{ className: "hidden" }}
+        centered
       >
-        <div className="content-overlay">
-          {error.isSuccess ? (
-            definitions.length !== 0 ? (
-              <ul>
-                {definitions.map((eachMeaning) => {
-                  return <li key={eachMeaning}>{eachMeaning}</li>;
-                })}
-              </ul>
-            ) : (
-              <div>Data loading...</div>
-            )
-          ) : (
-            <div>{error.errorMessage}</div>
-          )}
-        </div>
-      </div>
+        {wordDetails.sections ? (
+          <Collapse
+            defaultActiveKey={[wordDetails.sections[0]?.phonetic + "0"]}
+            accordion
+          >
+            {wordDetails.sections?.map((data, sectionIndex) => (
+              <Panel
+                header={
+                  <div className="main-collapse-title">
+                    {data.phonetic}
+                    <div>
+                      {partOfSpeechArray[sectionIndex].map(
+                        (partOfSpeech, partOfSpeechIndex) => (
+                          <Tag
+                            color="#777"
+                            key={
+                              sectionIndex + partOfSpeech + partOfSpeechIndex
```

Comment:
Lets follow a consistent format for keys.

` <label>-<index1>-<index2>..`

Make sure the indices are in the order of corresponding loops (the innermost index comes in the rightmost position). Also, use string interpolation.

Same for all keys.